src/main/java/org/apache/mesos/scheduler/repair/CoreFrameworkState.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.mesos.scheduler.repair;
+
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.state.StateStoreException;
+
+import java.util.List;
+
+/**
+ * Exposes simple task-oriented API for components that need to interact with the framework
+ * as a whole.
+ */
+public interface CoreFrameworkState {
```

Comment:
I don't understand what you mean by live Tasks.  The StateStore references the current state of Tasks.  Everything in this method is a getter except for deleteTask which I don't think should be here.